Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/Ratings/CQRS/Handlers/Create_RatingCommandHandler.cs

```diff
@@ -0,0 +1,77 @@
+﻿using AutoMapper;
+using BlogApp.Application.Contracts.Persistence;
+using BlogApp.Application.Exceptions;
+using BlogApp.Application.Features._Indices.CQRS.Commands;
+using BlogApp.Application.Features._Indices.DTOs.Validators;
+using BlogApp.Application.Features.Ratings.CQRS.Commands;
+using BlogApp.Application.Features.Ratings.DTOs.Validators;
+using BlogApp.Application.Responses;
+using BlogApp.Domain;
+using MediatR;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BlogApp.Application.Features.Ratings.CQRS.Handlers;
+public class Create_RatingCommandHandler : IRequestHandler<Create_RatingCommand, BaseResponse<int>>
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IMapper _mapper;
+
+    public Create_RatingCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
+    {
+        _unitOfWork = unitOfWork;
+        _mapper = mapper;
+    }
+    
+    public async Task<BaseResponse<int>> Handle(Create_RatingCommand request, CancellationToken cancellationToken)
+    {
+        var validator = new RatingDtoValidator();
+        var validationResult = await validator.ValidateAsync(request.RatingDto);
+        BaseResponse<int> response;
+        if (validationResult.IsValid == false)
+        {
+            response = new BaseResponse<int>()
+            {
+                Success = false,
+                Message = nameof(ValidationException),
+                Errors = validationResult.Errors.Select(error => error.ErrorMessage).ToList()
+            };
+            return response;
+        }
+        else
+        {
+            var rating = _mapper.Map<Rating>(request.RatingDto);
+            rating.BlogId = request.BlogId;
+            /* rating.raterId = _httpContextAccessor.HttpContext.User.Claims.FirstOrDefault(
```

Comment:
This won't be necessary if your are using the CreateRatingDto in your CreateRatingCommand.